src/middlewared/middlewared/plugins/system/product.py

```diff
@@ -218,6 +218,8 @@ def license_update(self, license_):
         SystemService.PRODUCT_TYPE = None
         if self.middleware.call_sync('system.is_enterprise'):
             Path('/data/truenas-eula-pending').touch(exist_ok=True)
```

Comment:
Instead of using `Path` to create the file, we should just do `with open(..., 'a+') as f:` Finally, we need to use `EULA_PENDING_PATH` var in this file.